frontend/src/lib/components/user-stats-modal.svelte

```diff
@@ -0,0 +1,135 @@
+<script lang="ts">
+  import { Button } from '$lib/components/ui/button';
+  import * as Dialog from '$lib/components/ui/dialog';
+  import { LoaderCircle } from 'lucide-svelte';
+  import type { ViewMatchDetailsV2, ViewUserDetails } from '../../api';
+  import Kpi from './kpi.svelte';
+  import type { LeaderboardEntry } from './leaderboard/leader-board-entry';
+  import MatchCard from './match-card/match-card.svelte';
+  import * as Card from './ui/card';
+
+  export let user: ViewUserDetails;
+  export let users: ViewUserDetails[];
+  export let leaderboardEntry: LeaderboardEntry | null | undefined;
+  export let open: boolean;
+  export let onOpenChange: (open: boolean) => void;
+
+  const percentFormatter = new Intl.NumberFormat(undefined, {
+    style: 'percent',
+    maximumFractionDigits: 0,
+  });
+
+  type RecentMatchDownload =
+    | {
+        status: 'success';
+        userId: number;
+        match: ViewMatchDetailsV2;
+      }
+    | {
+        status: 'error';
+        userId: number;
+        error: string;
+      }
+    | {
+        status: 'loading';
+        userId: number;
+      };
+
+  let recentMatch: RecentMatchDownload = {
+    status: 'loading',
+    userId: user.userId,
+  };
+
+  const fetchRecentMatch = async (userId: number) => {
+    recentMatch = {
+      status: 'loading',
+      userId,
+    };
+    try {
+      const response = await fetch(`/api/recent-match?playerId=${userId}`);
```

Comment:
Instead of calling fetch directly, why can't you use the auto generated api?

Reply:
It needs to go through the BFF since we don't have token in frontend

Reply:
And I would rather not just forward every single API request to the API completely unfiltered